### PR TITLE
Add support for Dataproc on GDC ApplicationEnvironment

### PR DESCRIPTION
**Pull request by JerryLeiDing**
This PR adds support for the Dataproc on GDC ApplicationEnvironment resource to the provider.

```release-note:new-resource
`google_dataproc_gdc_application_environment`
```

### CODE REVIEW

File: mmv1/products/dataprocgdc/ApplicationEnvironment.yaml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+name: ApplicationEnvironment
+references:
+  guides:
+    'Dataproc Intro': 'https://cloud.google.com/dataproc/'
+  api: 'https://cloud.google.com/dataproc-gdc/docs/reference/rest/v1/projects.locations.applicationEnvironments'
```
Comment (zli82016) [inline mmv1/products/dataprocgdc/ApplicationEnvironment.yaml:19]: This link returns 404

File: mmv1/products/dataprocgdc/ApplicationEnvironment.yaml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+self_link: projects/{{project}}/locations/{{location}}/serviceInstances/{{serviceinstance}}/applicationEnvironments/{{application_environment_id}}
+id_format: projects/{{project}}/locations/{{location}}/serviceInstances/{{serviceinstance}}/applicationEnvironments/{{application_environment_id}}
+import_format:
+  - projects/{{project}}/locations/{{location}}/serviceInstances/{{serviceinstance}}/applicationEnvironments/{{application_environment_id}}
+parameters:
+  - name: location
+    type: String
+    description: 'Resource ID segment making up resource `name`. It identifies the resource
+      within its parent collection. See documentation for resource type
+      `dataprocgdc.googleapis.com/ApplicationEnvironment`.'
```
Comment (zli82016) [inline mmv1/products/dataprocgdc/ApplicationEnvironment.yaml:32]: The description is not for `location`

File: mmv1/products/dataprocgdc/ApplicationEnvironment.yaml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    url_param_only: true
+    required: true
+    immutable: true
+  - name: serviceinstance
+    type: String
+    description: 'Resource ID segment making up resource `name`. It identifies the resource
+      within its parent collection. See documentation
```
Comment (zli82016) [inline mmv1/products/dataprocgdc/ApplicationEnvironment.yaml:39]: Can you add a link to documentation?

File: mmv1/third_party/terraform/services/dataprocgdc/resource_dataproc_gdc_application_environment_test.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  }
+  annotations = {
+    "an_annotation": "annotation_value"
+  }
+  spark_application_environment_config {
+    default_properties = {
+      "spark.executor.memory": "4g"
+    }
+    default_version = "1.2"
+  }
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/dataprocgdc/resource_dataproc_gdc_application_environment_test.go:91]: Can you update these fields with different values in `testAccDataprocGdcApplicationEnvironment_full`?

**Comment by JerryLeiDing** [comment]
@zli82016 The unit test failures don't look related to this PR, they are all occurring in providers which are not changed. Should we ignore these tests?

**Comment by zli82016** [comment]
> @zli82016 The unit test failures don't look related to this PR, they are all occurring in providers which are not changed. Should we ignore these tests?

You can ignore the failed unit test. Or rebase the main branch, which has a fix.

**Review by zli82016 — CHANGES_REQUESTED** [review]

File: mmv1/products/dataprocgdc/ApplicationEnvironment.yaml (hunk 2 above)
Comment (JerryLeiDing) [reply mmv1/products/dataprocgdc/ApplicationEnvironment.yaml:32]: Fixed the descriptions, these were created by Autogen and are not good descriptions for these fields

File: mmv1/products/dataprocgdc/ApplicationEnvironment.yaml (hunk 3 above)
Comment (JerryLeiDing) [reply mmv1/products/dataprocgdc/ApplicationEnvironment.yaml:39]: See above, fixed the description, these should not need documentation links

File: mmv1/third_party/terraform/services/dataprocgdc/resource_dataproc_gdc_application_environment_test.go (hunk 4 above)
Comment (JerryLeiDing) [reply mmv1/third_party/terraform/services/dataprocgdc/resource_dataproc_gdc_application_environment_test.go:91]: Done

File: mmv1/products/dataprocgdc/ApplicationEnvironment.yaml (hunk 1 above)
Comment (JerryLeiDing) [reply mmv1/products/dataprocgdc/ApplicationEnvironment.yaml:19]: The API docs are not published yet, but this is the location they will be hosted at once they are available

File: mmv1/third_party/terraform/services/dataprocgdc/resource_dataproc_gdc_application_environment_test.go [hunk 5]
```diff
@@ -0,0 +1,95 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/dataprocgdc/resource_dataproc_gdc_application_environment_test.go:2]: The copyright will be added automatically and not needed.

Comment (JerryLeiDing) [reply mmv1/third_party/terraform/services/dataprocgdc/resource_dataproc_gdc_application_environment_test.go]: Done, removed the copyright

**Review by zli82016 — APPROVED** [review]
LGTM

**Pull request merged**